Can't build project

Hi!
I can't compile project on CentOS release 6.6
g++ version:

```
g++ -v
Using built-in specs.
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-languages=c,c++,objc,obj-c++,java,fortran,ada --enable-java-awt=gtk --disable-dssi --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-1.5.0.0/jre --enable-libgcj-multifile --enable-java-maintainer-mode --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --disable-libjava-multilib --with-ppl --with-cloog --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.4.7 20120313 (Red Hat 4.4.7-18) (GCC)
```
error:
```
g++ -O2 -I/opt/jdk/jdk1.7.0_75//include -I/opt/jdk/jdk1.7.0_75//include/linux -fPIC -shared -o build/libasyncProfiler.so src/*.cpp -ldl -lpthread
In file included from src/profiler.h:22,
                 from src/javaApi.cpp:18:
src/spinLock.h: In member function ‘void SpinLock::spinLock()’:
src/spinLock.h:39: error: ‘__builtin_ia32_pause’ was not declared in this scope
In file included from src/perfEvent.h:23,
                 from src/perfEvent.cpp:25:
src/spinLock.h: In member function ‘void SpinLock::spinLock()’:
src/spinLock.h:39: error: ‘__builtin_ia32_pause’ was not declared in this scope
In file included from src/profiler.h:22,
                 from src/profiler.cpp:24:
src/spinLock.h: In member function ‘void SpinLock::spinLock()’:
src/spinLock.h:39: error: ‘__builtin_ia32_pause’ was not declared in this scope
In file included from src/profiler.h:22,
                 from src/vmEntry.cpp:22:
src/spinLock.h: In member function ‘void SpinLock::spinLock()’:
src/spinLock.h:39: error: ‘__builtin_ia32_pause’ was not declared in this scope
make: *** [build/libasyncProfiler.so] Error 1
```

--- a/src/perfEvent.cpp
+++ b/src/perfEvent.cpp
@@ -27,6 +27,17 @@
 
 #define rmb()  asm volatile("lfence":::"memory")
 
+// Ancient fcntl.h does not define F_SETOWN_EX constants and structures
+#ifndef F_SETOWN_EX
+#define F_SETOWN_EX  15
+#define F_OWNER_TID  0
+
+struct f_owner_ex {
+    int type;
+    pid_t pid;
+};
+#endif // F_SETOWN_EX
+
 
 int PerfEvent::_max_events = 0;
 PerfEvent* PerfEvent::_events = NULL;
@@ -39,7 +50,7 @@ void PerfEvent::init() {
 }
 
 int PerfEvent::tid() {
-    return syscall(SYS_gettid);
+    return syscall(__NR_gettid);
 }
 
 int PerfEvent::getMaxPid() {
@@ -65,7 +76,7 @@ void PerfEvent::createForThread(int tid) {
     attr.exclude_idle = 1;
     attr.precise_ip = 2;
 
-    int fd = syscall(SYS_perf_event_open, &attr, tid, -1, -1, 0);
+    int fd = syscall(__NR_perf_event_open, &attr, tid, -1, -1, 0);
     if (fd == -1) {
         perror("perf_event_open failed");
         return;

--- a/src/spinLock.h
+++ b/src/spinLock.h
@@ -17,6 +17,9 @@
 #ifndef _SPINLOCK_H
 #define _SPINLOCK_H
 
+#define x86_pause()  asm volatile("pause")
+
+
 // Cannot use regular mutexes inside signal handler
 class SpinLock {
   private:
@@ -36,7 +39,7 @@ class SpinLock {
 
     void spinLock() {
         while (!tryLock()) {
-            __builtin_ia32_pause();
+            x86_pause();
         }
     }
 

--- a/src/symbols.cpp
+++ b/src/symbols.cpp
@@ -148,7 +148,7 @@ bool ElfParser::loadSymbolsUsingBuildId() {
     }
 
     ElfNote* note = (ElfNote*)at(section);
-    if (note->n_namesz != 4 || note->n_descsz < 2 || note->n_descsz > 64 || note->n_type != NT_GNU_BUILD_ID) {
+    if (note->n_namesz != 4 || note->n_descsz < 2 || note->n_descsz > 64) {
         return false;
     }
 

